src/main/java/software/amazon/lambda/snapstart/ByteCodeIntrospector.java

```diff
@@ -43,7 +43,29 @@ private static Set<String> setOf(String ... strings) {
     };
 
     boolean isLambdaHandler(XClass xClass) {
-        return implementsLambdaInterface(xClass) || hasLambdaHandlerMethod(xClass);
+        return implementsLambdaInterface(xClass) || 
+               hasLambdaHandlerMethod(xClass) || 
+               (hasHandlerInClassName(xClass) && hasMethodCalledHandleRequest(xClass));
+    }
+
+    /**
+     * Returns true if the class has the word "handler" in the name.
+     */
+    private boolean hasHandlerInClassName(XClass xClass) {
+        return xClass.toString().contains("Handler") || xClass.toString().contains("handler");
```

Comment:
I'm worried that this might increase the noise significantly. Isn't there any other way to detect whether a class is auto-wired to a Lambda function?

Reply:
It's not really possible to do that. Basically customers can't use almost **any** class and method as a handler for a Lambda function (A very minimal example and a broader description in the issue related to this PR: #24). So we're following the most typical naming convention to catch more cases that we're not checking yet. And the condition here is a class name AND a specific method name, so hopefully that's enough of a restriction. 

The main problem here is that customers might not be using any of these as handlers, but they **could**, and there's no way of knowing that. (It's just a different `handler` in their function configuration and they start using it without changing anything in their code base)

Reply:
I would just keep `Handler` in upper case though, considering that it's a class name, you would expect CamelCasing, so "Handler" should start with a capital H.

Reply:
Got it, thanks. Updated.